lib/saml/SamlIdpUrlBuilder.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var njwt = require('njwt');
+var uuid = require('node-uuid');
+
+function SamlIdpUrlBuilder(application) {
+  this.application = application;
+  this.secret = application.dataStore.requestExecutor.options.client.apiKey.secret;
+}
+
+SamlIdpUrlBuilder.prototype._getServiceProvider = function _getServiceProvider(callback) {
+  this.application.getSamlPolicy(function (err, samlPolicy) {
+    if (err) {
+      return callback(err);
+    }
+
+    samlPolicy.getServiceProvider(callback);
+  });
+};
+
+SamlIdpUrlBuilder.prototype._buildInitializationUrl = function _buildInitializationUrl(initEndpointUrl, accessToken) {
+  return initEndpointUrl + '?accessToken=' + encodeURIComponent(accessToken);
+};
+
+SamlIdpUrlBuilder.prototype.build = function (callback) {
+  var self = this;
+
+  this._getServiceProvider(function (err, serviceProvider) {
+    if (err) {
+      return callback(err);
+    }
+
+    var accessToken = njwt.create({
```

Comment:
According to the [spec](https://stormpath.atlassian.net/wiki/display/AM/Stormpath+as+a+SAML+Service+Provider), the JWT header should include the API Key ID in the field `kid`.